packages/itwinui-react/src/core/Alert/Alert.tsx

```diff
@@ -132,16 +132,19 @@ AlertMessage.displayName = 'Alert.Message';
 
 const AlertAction = React.forwardRef((props, ref) => {
   const { children, className, ...rest } = props;
+  const { type } = useSafeContext(AlertContext);
 
   return (
-    <ButtonBase
+    <Anchor
```

Comment:
By making this change we lose out on the useful behaviors provided by `ButtonBase`.

I'm thinking we should use `Buttonbase` inside `Anchor` too. Or maybe we need a generic component called `Action` (or `ButtonOrLink`) 🤔

(Not sure of the best solution yet, and doesn't need to be resolved in this PR)

Reply:
I can test out using `ButtonBase` inside of the `Anchor`, and see if there are any issues with that.

Reply:
Not sure why did you change the `ButtonBase` to `Anchor` here. Would just adding `.iui-anchor` along with the necessary data attributes like how you did for `Toast` not be sufficient?

<img width="1008" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/4991e50f-8976-4b95-8c44-4b117ac9820c">

Reply:
`Anchor` is the correct way to do this. `Toast` should do the same

Reply:
The primary reason for making the change was to give access to the `underline` prop, which currently only exists on `Anchor`. The reason that `Toast` doesn't do the same is because the `Toast` anchor takes in a `link` prop, which has `HTMLButtonProps` as a part of it's type. Changing that would introduce a breaking change for the react package.

Reply:
Yeah lets handle this in a separate PR then. I think both Alert and Toast should correctly support both `<button>` and `<a>` because there are valid uses for both.

You can add an after-PR TODO in the PR description

Reply:
Added a TODO to the PR description.